src/rust_connection/fd_read_write.rs

```diff
@@ -76,54 +96,10 @@ pub trait WriteFD {
     }
 
     /// Flush this output stream, ensuring that all buffered contents are written out.
-    fn flush(&mut self) -> Result<()>;
-}
-
-/// Wraps a [`std::io::Write`] to implement the [`WriteFD`] trait.
-///
-/// Any attempts to write file descriptors will fail. Bytes are forwarded to the underlying writer.
-#[derive(Debug)]
-pub struct WriteFDWrapper<W: Write>(W);
-
-impl<W: Write> WriteFDWrapper<W> {
-    /// Create a new `WriteFDWrapper` for the given writer.
-    pub fn new(write: W) -> Self {
-        Self(write)
-    }
-}
-
-impl<W: Write> WriteFD for WriteFDWrapper<W> {
-    fn write(&mut self, buf: &[u8], fds: &mut Vec<RawFdContainer>) -> Result<usize> {
-        check_no_fds(fds)?;
-        self.0.write(buf)
-    }
-
-    fn write_all(&mut self, bufs: &[u8], fds: Vec<RawFdContainer>) -> Result<()> {
-        check_no_fds(&fds)?;
-        self.0.write_all(bufs)
-    }
-
-    fn write_vectored(
-        &mut self,
-        bufs: &[IoSlice<'_>],
-        fds: &mut Vec<RawFdContainer>,
-    ) -> Result<usize> {
-        check_no_fds(&fds)?;
-        self.0.write_vectored(bufs)
-    }
-
-    fn flush(&mut self) -> Result<()> {
-        self.0.flush()
-    }
-}
-
-// Check that the given argument is an empty slice
-fn check_no_fds(fds: &[RawFdContainer]) -> Result<()> {
-    if !fds.is_empty() {
-        Err(Error::new(ErrorKind::Other, "FD passing is unsupported"))
-    } else {
-        Ok(())
-    }
+    /// `block` allows to specify whether this method shall block or not. If `block` is
```

Comment:
Is this necessary? The only caller for blocking flushing is `write_setup()`. Would it be possible to just use `poll` there instead and then use the non-blocking flushing?
And if the answer is "yes": Would that be worth it? Or is it better to have blocking/nonblocking flush variant?

Reply:
Changed, `flush` is now always non-blocking. I also removed `write_all` because it was only used in `write_setup` too.